### PR TITLE
Fix #2237: Allow registering argument types without interacting with the class map

### LINKED ISSUE #2237: Allow registering argument types without interacting with the class map
In Minecraft 1.18.2, argument types can be registered to using a helper method:

```java
ArgumentTypes.register(id.toString(), CustomArgumentType.class, serializer);
```

In Minecraft 1.19, registering an argument type is more difficult than simply registering to the registry because the class map (a private field) must also be modified:

```java
Registry.register(Registry.COMMAND_ARGUMENT_TYPE, id, serializer);
ArgumentTypesAccessor.getClassMap().put(CustomArgumentType.class, serializer);
```

Ideally, Fabric API should provide a helper method that registers an argument type without requiring a stringified identifier or private access to `ArgumentTypes.CLASS_MAP`.

### PR DESCRIPTION
**Pull request by Technici4n**
Fixes #2237.

### CODE REVIEW

**Comment by modmuss50** [comment]
A test mod would be very useful imo.

**Comment by modmuss50** [comment]
Any chance of adding a testmod to this? Or shall I just merge it anyway?

**Comment by Technici4n** [comment]
I'll add a test mod don't worry. Before 1.19 for sure.

**Comment by Technici4n** [comment]
Testmod is added and it works! 😄

image
image
image
image
image

**Review by modmuss50 — APPROVED** [review]

File: fabric-command-api-v2/src/testmod/java/net/fabricmc/fabric/test/command/CustomArgumentTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	@Override
+	public void onInitialize() {
+		ArgumentTypeRegistry.registerArgumentType(new Identifier("fabric-command-test", "smiley"), SmileyArgumentType.class, ConstantArgumentSerializer.of(SmileyArgumentType::smiley));
+
+		CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> {
+			dispatcher.register(
+					literal("fabric_custom_argument_test").then(
+							argument(ARG_NAME, SmileyArgumentType.smiley())
+								.executes(CustomArgumentTest::executeSmileyCommand)));
+		});
```
Comment (haykam821) [inline fabric-command-api-v2/src/testmod/java/net/fabricmc/fabric/test/command/CustomArgumentTest.java:47]: Could this command be registered as a client command as well? Testing `registryAccess` would be useful for both sides.

Comment (Technici4n) [reply fabric-command-api-v2/src/testmod/java/net/fabricmc/fabric/test/command/CustomArgumentTest.java:47]: This command doesn't test the registryAccess anyway?

Comment (haykam821) [reply fabric-command-api-v2/src/testmod/java/net/fabricmc/fabric/test/command/CustomArgumentTest.java:47]: It should. The smiley enum could potentially become a registry.

Comment (Technici4n) [reply fabric-command-api-v2/src/testmod/java/net/fabricmc/fabric/test/command/CustomArgumentTest.java:47]: Let's maybe try to keep the testmod reasonable... 😅

**Review by Player3324 — APPROVED** [review]

**Pull request merged**